lib/transformers/typescript.js

```diff
@@ -16,6 +52,21 @@ module.exports = {
     const tsconfig = getTsJestConfig(config)
     const babelOptions = getBabelOptions(filePath)
 
+    if (vueJestConfig.enableExperimentalTsDiagnostics) {
+      const diagnostics = getDiagnostics(
+        scriptContent,
+        { ...tsconfig, noEmitOnError: true },
+        typescript
+      )
+      logDiagnostics(filePath, diagnostics, typescript)
+
+      if (diagnostics.length) {
```

Comment:
You can put warning message here, that is how currently ts-jest [does](https://github.com/kulshekhar/ts-jest/blob/452639242d7f59502bda7aa94b81735f9c9ad203/src/config/config-set.ts#L382) with `isolatedModules: true`